qutebrowser/browser/network/networkmanager.py

```diff
@@ -295,6 +297,25 @@ def adopt_download(self, download):
         download.destroyed.connect(self.on_adopted_download_destroyed)
         download.do_retry.connect(self.adopt_download)
 
+    def setReferer(self, req, current_url):
+        """Set the referer header."""
+        referer_header_conf = config.get('network', 'referer-header')
+
+        try:
+            if referer_header_conf == 'never':
+                # Note: using ''.encode('ascii') sends a header with no value,
+                # instead of no header at all
+                req.setRawHeader('Referer'.encode('ascii'), QByteArray())
+            elif (referer_header_conf == 'same-domain' and
+                    not urlutils.same_domain(req.url(), current_url)):
+                req.setRawHeader('Referer'.encode('ascii'), QByteArray())
+            # If refer_header_conf is set to 'always', we leave the header
+            # alone as QtWebKit did set it.
+        except urlutils.InvalidUrlError:
+            # req.url() or current_url can be invalid - this happens on
+            # https://www.playstation.com/ for example.
+            pass
+
```

Comment:
Ah, sorry for the confusion - just saw the answer on the commit comment. Either way I'd be glad if you could rename this, and then it's ready for merging (but I'd like to get the tests running for you first).